src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -430,24 +430,21 @@ static byte[] generateDirectMethodHandleHolderClassBytes(String className,
             names.add(form.kind.defaultLambdaName);
         }
         for (Wrapper wrapper : Wrapper.values()) {
-            if (wrapper == Wrapper.VOID) {
-                continue;
-            }
+            int ftype = wrapper == Wrapper.VOID ? DirectMethodHandle.FT_CHECKED_REF : DirectMethodHandle.ftypeKind(wrapper.primitiveType());
             for (byte b = DirectMethodHandle.AF_GETFIELD; b < DirectMethodHandle.AF_LIMIT; b++) {
-                int ftype = DirectMethodHandle.ftypeKind(wrapper.primitiveType());
                 LambdaForm form = DirectMethodHandle
                         .makePreparedFieldLambdaForm(b, /*isVolatile*/false, ftype);
-                if (form.kind != LambdaForm.Kind.GENERIC) {
-                    forms.add(form);
-                    names.add(form.kind.defaultLambdaName);
-                }
+                if (form.kind == GENERIC)
+                    throw new InternalError(b + " non-volatile " + ftype);
+                forms.add(form);
+                names.add(form.kind.defaultLambdaName);
```

Comment:
I suppose the fact that static/non-static variants have the same name is okay since they will have a different number of parameters?

Reply:
Yep, I left the comment at https://github.com/openjdk/jdk/pull/24887/files#diff-62978b9d88b0791f31c654103e6b97eea8bf711fa37ac4bf614847b26f44e0c1R761